build.sh

```diff
@@ -45,21 +70,32 @@ done
 
 mkdir -p logs
 
+# Check out and patch Mono version
+if [ ! -e ${mono_root} ]; then
+    git clone https://github.com/mono/mono ${mono_root} --branch ${mono_version} --single-branch --progress --depth 1 
+    pushd ${mono_root}
+    if [ ! -z "${mono_commit}" ]; then 
+	git checkout ${mono_commit} 
+    fi
+    git submodule update --init --recursive --recommend-shallow --progress --depth 1 
+    patch -p1 < ${files_root}/patches/mono-unity-Clear-TLS-instead-of-aborting.patch 
+    popd
+fi
+
```

Comment:
Download mono plus 25+ submodules once instead of 7x! Also only need shallow depth, but I don't think it saved much.

Reply:
Makes sense. @hpvb's idea was to use a build stage for Mono instead to reuse the same stage in all containers without making them too big (which would happen if we closed Mono in the base container and just kept it there for downstream builds). But we can still do that later on, this is a good stopgap solution.

Reply:
Does `--depth 1` still allow us to checkout a commit which might be older than `HEAD` though?

Reply:
No. You're right, that will break it. I had to remove that from the other dockers that checkout. Will remove that.

Reply:
The `[ ! -e ${mono_root} ]` will also prevent checking out a different commit in the same branch, since only the branch name is part of the `mono_root`.